File: CitasAPI/src/main/java/citas_service_nuevo/repository/CitaRepository.java
package citas_service_nuevo.repository;

import citas_service_nuevo.model.Cita;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDateTime;
import java.util.List;

@Repository
public interface CitaRepository extends JpaRepository<Cita, Long> {

    List<Cita> findByIdUsuario(Long idUsuario);

    List<Cita> findByIdUsuarioAndFechaCitaAfter(Long idUsuario, LocalDateTime fecha);

    List<Cita> findByIdDoctorAndFechaCitaBetweenOrderByHoraInicio(Long idDoctor, LocalDateTime fechaInicio, LocalDateTime fechaFin);

    List<Cita> findByIdDoctorAndFechaCitaAfter(Long idDoctor, LocalDateTime fecha);
}

File: citas-service-nuevo/src/main/kotlin/citas_service_nuevo/repository/CitaRepository.kt
package citas_service_nuevo.repository

import citas_service_nuevo.model.Cita
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
interface CitaRepository : JpaRepository<Cita, Long> {

    fun findByUsuarioId(idUsuario: Long): List<Cita>

    // El metodo se llama findByUsuarioIdAndFechaCitaAfter
    fun findByUsuarioIdAndFechaCitaAfter(idUsuario: Long, fecha: LocalDateTime): List<Cita>
}